migrations/2_setup_ens.js

```diff
@@ -0,0 +1,38 @@
+const ENS = artifacts.require('@ensdomains/ens/ENSRegistry');
+const FIFSRegistrar = artifacts.require('@ensdomains/ens/FIFSRegistrar');
+const PublicResolver = artifacts.require('@ensdomains/resolver/PublicResolver');
+
+const utils = require('web3-utils');
+const namehash = require('eth-ens-namehash');
+
+const TLD = 'test';
+
+// const ENS_MAINNET = 0x314159265dd8dbb310642f98f50c066173c1259b;
+// const ENS_ROPSTEN = 0x112234455c3a32fd11230c42e7bccd4a84e02010;
+// const ENS_RINKEBY = 0xe7410170f87102df0055eb195163a03b7f2bff4a;
+
+module.exports = async (deployer, network, accounts) => {
+  await deployer.deploy(ENS);
+  const ens = await ENS.deployed();
+
+  await deployer.deploy(PublicResolver, ens.address);
```

Comment:
Are we supposed to be redeploying the public resolver?

Reply:
Just out of curiosity, what is the reasoning behind using our own ENSRevolver over the PublicResolver?

Reply:
To reiterate what I said on slack PM, the reason I went for this was because I was having trouble import the artifacts from the package, I could however import the solidity files from the package and recompile but PublicResolver was a pain to compile due to some external dependencies not self contained in the package, those external dependencies would also need recompiling and some of them need a different compiler version, I decided since we only need a resolver and not the public resolver it would be simplier to just design a basic resolver that fits our purposes and deploy that